internal/hooks/hook_executor_v2.go

```diff
@@ -38,7 +38,7 @@ type HookExecutorMessageBoundaryProtocol struct {
 var generateBoundary = generateMD5FromRandomString
 
 // Execute processes the data received by the SDK.
-func (e *HookExecutorMessageBoundaryProtocol) Execute(opts HookExecOpts) (string, error) {
+func (e *HookExecutorMessageBoundaryProtocol) Execute(ctx context.Context, opts HookExecOpts) (string, error) {
```

Comment:
⭐ ⭐ ⭐ 
note: This is 2 of 2 key changes in this PR. Here we update the V2 hook executor protocol to accept a context argument. This allows us to remove the `context.Background()` usage inside this function.
⭐ ⭐ ⭐